packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -36,6 +36,13 @@ export const DataTable = <T extends DomainObject>({
     if (data.length) setActiveRows(data.map(({ id }) => id as string));
   }, [data]);
 
+  // guard against a page number being set which is greater than the data allows
+  useEffect(() => {
+    if (!pagination || !onChangePage || !pagination.total) return;
+    const { page, first, total } = pagination;
+    if (page * first > total) onChangePage(0);
+  }, [pagination]);
```

Comment:
annoying that `total` isn't in the `usePagination` or `useQueryParams` hooks :(

Reply:
yup - that would have made it neater!